### ISSUE TITLE
the gpu kernel of cross entropy with soft label can be optimized by using reduce.

### ISSUE THREAD

**Issue by lcy-seso** [PaddlePaddle/Paddle#4236]
The current implementation of the GPU kernel of the cross_entropy op with the soft label can be further optimized by using reduce:
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/cross_entropy_op.cu#L38

This problem is also known by @qingqing01

**Comment by typhoonzero** [comment]
I used `thrust::reduce` for convenience, it's very simple to use and efficient: https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/accuracy_op.cu#L43

**Comment by lcy-seso** [comment]
Thank you ~ @typhoonzero
